Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -180,6 +233,37 @@ public TimeSpan UpdateState(string instrumentationKey)
             return this.DetermineBackOffs();
         }
 
+        private void UpdateConfiguration(CollectionConfigurationInfo configurationInfo)
+        {
+            // we only get here if Etag in the header is different from the current one, but we still want to check if Etag in the body is also different
+            if (configurationInfo != null && !string.Equals(configurationInfo.ETag, this.currentConfigurationETag, StringComparison.Ordinal))
+            {
+                this.collectionConfigurationErrors.Clear();
```

Comment:
This is not the best pattern for applying the configuration. Is there any reason to not use "snapsots" for configuration rather than messing up with instantiated config and track state changes?